### PR TITLE
Modernize to c++11

### PR DESCRIPTION
**Pull request by hokein**
This PR has no functional change. Modernize the C++03 code to C++11 (all the features are allowed in Chromium code style). Automatically converted by `modernize-*` checks in clang-tidy. Tested on macOS locally.

### CODE REVIEW

File: atom/browser/api/atom_api_session.cc [hunk 1]
```diff
@@ -241,10 +241,10 @@ void OnGetBackend(disk_cache::Backend** backend_ptr,
     } else if (action == Session::CacheAction::STATS) {
       base::StringPairs stats;
       (*backend_ptr)->GetStats(&stats);
-      for (size_t i = 0; i < stats.size(); ++i) {
-        if (stats[i].first == "Current size") {
+      for (auto& stat : stats) {
```
Comment (zcbenz) [inline atom/browser/api/atom_api_session.cc]: We should probably always prefer `const auto&` in ranged for loops.

File: atom/common/node_bindings_mac.cc [hunk 2]
```diff
@@ -54,7 +54,7 @@ void NodeBindingsMac::PollEvents() {
   // Wait for new libuv events.
   int r;
   do {
-    r = select(fd + 1, &readset, NULL, NULL, timeout == -1 ? NULL : &tv);
+    r = select(fd + 1, &readset, nullptr, nullptr, timeout == -1 ? nullptr : &tv);
```
Comment (zcbenz) [inline atom/common/node_bindings_mac.cc]: This line exceeds 80 columns.

**Comment by zcbenz** [comment]
Can you keep the files under `chromium_src` untouched? Changing them would make it much harder to sync those files with upstream Chromium codebase.

File: atom/browser/api/atom_api_session.cc (hunk 1 above)
Comment (hokein) [inline atom/browser/api/atom_api_session.cc]: Done.

File: atom/common/node_bindings_mac.cc (hunk 2 above)
Comment (hokein) [inline atom/common/node_bindings_mac.cc]: Done.

**Comment by hokein** [comment]
> Can you keep the files under chromium_src untouched? Changing them would make it much harder to sync those files with upstream Chromium codebase.

Done.

**Comment by paulcbetts** [comment]
@hokein This is good, can you write up a doc of what people should do in the future to follow these guidelines?

**Comment by zcbenz** [comment]
Thanks!

**Comment by Slamology2** [comment]
Location: Can't find: it on.
ftp://ftp.www.map.com
Try. Skipping to Mechanic Desktop

**Comment by hokein** [comment]
@paulcbetts 
Basically, we can use most C++11 features when writing C++ code now. The features are listed in https://chromium-cpp.appspot.com/. It makes sense to me to update the electron code style document.

**Pull request merged**